### DISCUSSION TITLE
Backend Logic Idea for Kanoodle solver.

### DISCUSSION THREAD

**Discussion by Rez27** [thepercyj/Puzzle-Realms#15] [Ideas]
Hi Team,
Sorry for posting this late but here i'm sharing all the possible solutions which we got for implementing the DLX- dancing links algorithm for our kanoodle problem so far.
The 2 different solution sets are categorized as follows

- **Solution 1** - With no rotations/reflections - 3,71,020 in 221827 ms. (_Implemented in Python by Brute Force_)
- **Solution 2** - With rotations/reflections - 43,31,140 in 2386416 ms. (_Implemented in Java by DLX Algo_)

Now working on the implementing the DLX algo with rotations in python in order to integrate it into our Djano App.

@thepercyj Thanking you for collectively working on the Java code. 
@JUNKAIOU Thanks for running the Brute force algorithm. 

We now have what we can call is a benchmark for all the possible number of solutions with each of the approaches. Need further work on these codes to implement them in function with variable inputs for our Interactive Kanoodle solver

**Comment by thepercyj** [comment]
Hi Team,
We've had some updates to our logic. Our logic was missing a partial configuration solver, which would take all the pieces as input and display solutions. So, we added an initialized board function that will pre-configure some pieces and based on that it will solve the remaining pieces inside the grid.

Now working on the stress testing for all puzzle pieces to see if it work properly or not

@Rez27 : Thanking collectively for helping solve the logic and providing matrix board to visualize the pieces 
@tnaccarato: Thanking collectively for helping solve the logic and providing additional puzzle pieces for validation
@Mya2328: Thanking collectively for helping solve the logic and spilling some ideas.
@jawadnasar: Thanking collectively for helping solve the logic and providing routine checks for the timeline of our project process.

**Comment by thepercyj** [comment]
Hi @tnaccarato,

Just updated the django views and models for it, seems some imports and function are still not working. Can you pick it up for me ?. Thanks.

**Comment by tnaccarato** [comment]
Does anyone know anything about websockets? I thought it might be nice to update the page as solutions are being found because it is taking forever to find the solutions and just looks like the page isn't doing anything while it calculates them all! We probably won't have time to do anything about it now as using asynchronous processing looks pretty involved but just a thought

**Reply by jawadnasar** [reply]
Yes I know about them but I can implement in other language not the one we are working on right now
